src/Umbraco.Core/Persistence/IUmbracoDatabaseFactory.cs

```diff
@@ -42,6 +42,12 @@ public interface IUmbracoDatabaseFactory : IDisposable
         /// </summary>
         bool CanConnect { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether the database is empty, it should already be determined if its possible to
+        /// connect to the database (see <see cref="CanConnect"/>) otherwise an exception might be thrown.
+        /// </summary>
+        bool IsDatabaseEmpty { get; }
```

Comment:
This is a breaking change. This could just be an ext method right? Since all it's doing is:

```cs
 using (var database = dbFactory.CreateDatabase())
 {
      return SqlContext.SqlSyntax.GetTablesInSchema(database).Any() == false;
 }
```

Reply:
Yeah it could, was just thinking it were nice to have it in a shared place.

Reply:
Moved to `RuntimeState.cs`, reason for adding the check is that `EnsureUmbracoUpgradeState` could throw of other reasons than the database is empty